### ISSUE TITLE
Problem with migrating from old tantan-s3-cloudfront plugin

### ISSUE THREAD

**Issue by Shebo** [deliciousbrains/wp-amazon-s3-and-cloudfront#401]
Hi there,
I have big wpmu site with 1,000,000+ images stored in S3, WP is originally installed as 2.9/3.1 version and upgraded since.
Up until now the website worked with an old-modified version of this plugin:
https://wordpress.org/plugins/tantan-s3-cloudfront/

I want to migrate to the latest WP Offload S3 Lite, but I cannot make the url align with the existing format.
The current folder structure is like so:
`http://test.images.cloudfront.com/main_folder/project_folder/files/2017/10/image.png` - for main blog media
`http://test.images.cloudfront.com/main_folder/project_folder/blog_name/files/2017/10/image.png` - for sub-blogs media.

Now, when I'm replacing that old plugin with WP Offload S3 Lite, the url structure is all messy. sub-blogs are getting this folder structure:
`http://test.images.cloudfront.com/sites/8908/2017/10/image.png`

while main blog get messy ABSPATH folder structure (from a local dev copy on a windows machine):
`http://test.images.cloudfront.com/C:xampphtdocsmain_folderproject_folder/wp-content/blogs.dir/1/files/2017/10/image.png`.

Now, i'm aware that these issues that are the result of upgrading an old wpmu (like presented in #255 ), with db mismatches and old rewrite rules and whatnot, but the reality is that I can't drop support at the already-existed S3 assets or the old upgraded DB.

I really want to advance to an updated plugin with better handling and not get stuck with the old unsupported one.

Have you got any suggestions on how can I accomplish that?
Could you maybe add a filter to the prefix creation process (maybe in get_dynamic_prefix or get_file_prefix methods) so I can hook in and change the folder structure to my needs?

Thank you for reading so far, I'd appreciate any kind of help on this subject.

**Comment by ianmjones** [comment]
Hi @Shebo,

As I'm sure you can understand, you're asking for support for migrating from one very old free plugin that is the predecessor to another free plugin, on a WordPress site that likely has many edge cases due to its age and size.

However, you may be able to achieve what you're looking for by implementing either the `as3cf_get_attachment_url` (most likely) or `as3cf_wp_get_attachment_url` filters. Check out the [Tweaks plugin](https://github.com/deliciousbrains/wp-amazon-s3-and-cloudfront-tweaks/blob/04e7c00fd62bc13620ae7185e926ad80293fc3f3/amazon-s3-and-cloudfront-tweaks.php#L216) for examples.

For further support, please raise a request at https://wordpress.org/support/plugin/amazon-s3-and-cloudfront or if you have a WP Offload S3 license please use the "License & Support" tab in the plugin.

**Comment by Shebo** [comment]
Thank you for the response, @ianmjones.

I already tried those filters, but too much things (bucket key, bucket folder etc) were out of sync.
I ended finding out that my wpmu wasn't fully upgraded and compatible with the latest wp versions.

After making sure it was fully upgraded, I activated the plugin and it worked as expected, and only needed a little bit of filter tweaking to make it work as I needed.
